packages/flutter_tools/lib/src/isolated/web_asset_server.dart

```diff
@@ -0,0 +1,777 @@
+    logging.Logger.root.onRecord.listen((logging.LogRecord event) => log(logger, event));
+
+    // Retrieve connected web devices.
+    final List<Device>? devices = await globals.deviceManager?.getAllDevices();
+    final Set<String> connectedWebDeviceIds =
+        devices
+            ?.where((Device d) => d.platformType == PlatformType.web && d.isConnected)
+            .map((Device d) => d.id)
+            .toSet() ??
+        <String>{};
+
+    // In debug builds, spin up DWDS and the full asset server.
+    final Dwds dwds = await dwdsLauncher(
+      assetReader: server,
+      buildResults: const Stream<BuildResult>.empty(),
+      chromeConnection: () async {
+        final Chromium chromium = await chromiumLauncher!.connectedInstance;
+        return chromium.chromeConnection;
+      },
+      toolConfiguration: ToolConfiguration(
+        loadStrategy:
+            ddcModuleSystem
+                ? FrontendServerDdcLibraryBundleStrategyProvider(
+                  ReloadConfiguration.none,
+                  server,
+                  PackageUriMapper(packageConfig),
+                  digestProvider,
+                  BuildSettings(
+                    appEntrypoint: packageConfig.toPackageUri(
+                      fileSystem.file(entrypoint).absolute.uri,
+                    ),
+                  ),
+                  packageConfigPath: buildInfo.packageConfigPath,
+                  hotReloadSourcesUri: server._baseUri!.replace(
+                    pathSegments: List<String>.from(server._baseUri!.pathSegments)
+                      ..add(_reloadScriptsFileName),
+                  ),
+                ).strategy
+                : FrontendServerRequireStrategyProvider(
+                  ReloadConfiguration.none,
+                  server,
+                  PackageUriMapper(packageConfig),
+                  digestProvider,
+                  BuildSettings(
+                    appEntrypoint: packageConfig.toPackageUri(
+                      fileSystem.file(entrypoint).absolute.uri,
+                    ),
+                  ),
+                  packageConfigPath: buildInfo.packageConfigPath,
+                ).strategy,
+        debugSettings: DebugSettings(
+          enableDebugExtension: true,
+          urlEncoder: urlTunneller,
+          useSseForDebugProxy: useSseForDebugProxy,
+          useSseForDebugBackend: useSseForDebugBackend,
+          useSseForInjectedClient: useSseForInjectedClient,
+          expressionCompiler: expressionCompiler,
+          spawnDds: enableDds,
+        ),
+        appMetadata: AppMetadata(hostname: hostname),
+      ),
+      // Inject the debugging support code if connected web devices are present,
+      // and user specified a device id that matches a connected web device.
+      // If the user did not specify a device id, we use chrome as the default.
+      injectDebuggingSupportCode:
+          connectedWebDeviceIds.isNotEmpty &&
```

Comment:
This line can be removed as `Set.contains(...)` will return false if the set is empty.